### ISSUE TITLE
Warning: setState(...): Can only update a mounted or mounting component. 

### ISSUE THREAD

**Issue by t1mmen** [ameyms/react-animated-number#7]
Hi,

First off, thanks for an awesome lib.

We've experienced some issues when the animation is running but the component gets unmounted, as seen here:

![screenshot 2016-11-15 17 09 52](https://cloud.githubusercontent.com/assets/192263/20330841/56ab08de-ab56-11e6-90dc-05250253c2fe.png)

I believe this is related to `tweenValue`'s (possibly also `endTween`'s) `setState` call.

Probably workaround:

```jsx
componentDidMount() {
    this.setState({ isMounted: true });
}
componentWillUnmount() {
    this.setState({ isMounted: false}, () => {
        this.endTween();
    });
}
// ....


// ... not sure if needed
endTween() {
    raf.cancel(this.tweenHandle);
    if (this.state.isMounted) {
        this.setState({
            ...this.state,
            currentValue: this.props.value
        });
    }
}

// ... but this probably is:
tweenValue(timestamp, start) {
    // ....
    if (this.state.isMounted) {
        this.setState({
            currentValue: newValue,
            startTime: startTime ? startTime : currentTime,
            fromValue, currentTime
        });
    }
    // ...
}
```

**Comment by ameyms** [comment]
Thanks @t1mmen. 
👾  It will be helpful if you put together a small gist to reproduce this

⚠️  This is a duplicate of #3 but keeping this open since the other is closed.

My initial guess this could be fixed if we update `tweenHandle` [every](https://github.com/ameyms/react-animated-number/blob/master/src/AnimatedNumber.jsx#L100) [time](https://github.com/ameyms/react-animated-number/blob/master/src/AnimatedNumber.jsx#L132) we invoke `raf`.
Perhaps pass all calls to `raf()` through a private method.

👉   I am assigning this to myself but PR is most welcome.

**Comment by septs** [comment]
https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html

**Comment by t1mmen** [comment]
@ameyms I've been unable to pin down exactly when/why this happens, I'm afraid. If I manage to figure it out, I'll produce a gist for you.

@septs That is why I didn't check `this.isMounted()` but kept it in state, though a property on the class is probably better as described in the link you pasted:

> An easy migration strategy for anyone upgrading their code to avoid `isMounted()` is to track the mounted status yourself. Just set a `_isMounted` property to `true` in `componentDidMount` and set it to `false` in `componentWillUnmount`, and use this variable to check your component's status.

**Comment by ameyms** [comment]
Thanks @t1mmen and @septs for chiming in.

Like I have said in my previous comment, I know what the potential fix is:
> My initial guess this could be fixed if we update tweenHandle every time we invoke raf.
Perhaps pass all calls to raf() through a private method.

Without clear steps to reproduce, I want to go with this guess since it is something that needs to be fixed anyways.
I will push an update this coming Saturday.

**Comment by idw111** [comment]
how to reproduce error
- pass long enough duration for AnimatedNumber component
- transition to another page while animating (to make the AnimatedNumber unmounted)
- error occurs

**Comment by bcole808** [comment]
+1 as I am also having this issue. As @idw111 said, this occurs when you:

1. Create an `<AnimatedNumber>` component with a long animation duration (i.e. 10 seconds)
2. Cause the `<AnimatedNumber>` to be removed from the page before the animation completes (perhaps by changing state of the parent component, such that the animated number is no longer displayed). 
3. See state warning in console because Animated Number is still trying to complete the animation even though it was removed. 

Using version 0.4.2

**Comment by phiresky** [comment]
Fixed in #13
